src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,28 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_LENGTH_OF_LOGIN_AND_PASSWORD = 6;
+    private static final int MINIMUM_USER_AGE_TO_LOGIN = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidDataException("User should be not null");
+        }
+        if (storageDao.get(user.getLogin()) == null
+                && user.getLogin().length() >= MINIMUM_LENGTH_OF_LOGIN_AND_PASSWORD
+                && user.getPassword().length() >= MINIMUM_LENGTH_OF_LOGIN_AND_PASSWORD
+                && user.getAge() >= MINIMUM_USER_AGE_TO_LOGIN) {
+            storageDao.add(user);
```

Comment:
What if password is null? devide this logic for checking every parameter in separate method

Reply:
Changed.